doc/rados/operations/crush-map.rst

```diff
@@ -3,103 +3,103 @@
 ============
 
 The :abbr:`CRUSH (Controlled Replication Under Scalable Hashing)` algorithm
-determines how to store and retrieve data by computing storage locations.
-CRUSH empowers Ceph clients to communicate with OSDs directly rather than
-through a centralized server or broker. With an algorithmically determined
+computes storage locations in order to determine how to store and retrieve
+data.  CRUSH allows Ceph clients to communicate with OSDs directly rather than
+through a centralized server or broker. By using an algorithmically-determined
 method of storing and retrieving data, Ceph avoids a single point of failure, a
 performance bottleneck, and a physical limit to its scalability.
 
-CRUSH uses a map of your cluster (the CRUSH map) to pseudo-randomly
-map data to OSDs, distributing it across the cluster according to configured
-replication policy and failure domain.  For a detailed discussion of CRUSH, see
+CRUSH uses a map of the cluster (the CRUSH map) to pseudorandomly map data to
+OSDs, distributing the data across the cluster with reference to configured
+replication policy and failure domain. For a detailed discussion of CRUSH, see
 `CRUSH - Controlled, Scalable, Decentralized Placement of Replicated Data`_
 
 CRUSH maps contain a list of :abbr:`OSDs (Object Storage Devices)`, a hierarchy
-of 'buckets' for aggregating devices and buckets, and
-rules that govern how CRUSH replicates data within the cluster's pools. By
-reflecting the underlying physical organization of the installation, CRUSH can
-model (and thereby address) the potential for correlated device failures.
-Typical factors include chassis, racks, physical proximity, a shared power
-source, and shared networking. By encoding this information into the cluster
-map, CRUSH placement
-policies distribute object replicas across failure domains while
-maintaining the desired distribution. For example, to address the
-possibility of concurrent failures, it may be desirable to ensure that data
-replicas are on devices using different shelves, racks, power supplies,
-controllers, and/or physical locations.
-
-When you deploy OSDs they are automatically added to the CRUSH map under a
-``host`` bucket named for the node on which they run.  This,
-combined with the configured CRUSH failure domain, ensures that replicas or
-erasure code shards are distributed across hosts and that a single host or other
-failure will not affect availability.  For larger clusters, administrators must
-carefully consider their choice of failure domain.  Separating replicas across racks,
-for example, is typical for mid- to large-sized clusters.
+of 'buckets' for aggregating devices and S3 buckets, and rules that govern how
```

Comment:
Ceph is rife with overloaded nomenclature.

S3 has buckets that are more or less directories.  These are different.

The buckets here are CRUSH buckets -- OSDs, hosts, chassis, racks, rows, rooms, datacenters, etc.  Any type of node (there's another overload!) in the CRUSH hierarchy graph.

so I suggest

>list of OSDs and a hierarchy of "buckets" (``host``s, ``rack``s, etc.) and rules that govern how

Reply:
Accepted.